seed/utils/filter_state.py

```diff
@@ -0,0 +1,175 @@
+"""
+:copyright (c) 2014 - 2022, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db.utils import DataError
+from django.http import JsonResponse
+from rest_framework import status
+from rest_framework.request import Request
+from seed.lib.superperms.orgs.models import Organization
+from seed.models import (VIEW_LIST, VIEW_LIST_PROPERTY, VIEW_LIST_TAXLOT,
+                         Column, ColumnListProfile, ColumnListProfileColumn,
+                         Cycle, PropertyView)
+from seed.models import TaxLotProperty, TaxLotView
+from seed.search import build_view_filters_and_sorts, FilterException
+from seed.serializers.pint import (apply_display_unit_preferences)
+from typing import Union, Optional, Literal
+
+
+def _get_filtered_results(request: Request, state_type: Union[Literal['property'], Literal['taxlot'], None], profile_id: int):
```

Comment:
Will have to import HttpResponse from django.http -- I think this makes it a bit clearer that it's fully handling the HTTP response as well as filtering.

I think we should be a bit more general about the `state_type` name since it's really filtering on *Views, so maybe use `inventory_type` instead

I removed the `Union` with `None` -- I'm not sure if we want to allow users to pass None here right? What would that mean? I also simplified the `Literal`s, you can list multiple literals in the annotation

Also, I tweaked the function name. I removed the leading `_` b/c it's not really a private function -- we're using it in other files. Also I think a more general name (inventory) instead of state might be clearer b/c this is technically filtering for *Views

```suggestion
def get_filtered_inventory(request: Request, inventory_type: Literal['property', 'taxlot'], profile_id: int) -> HttpResponse:
```